flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -268,6 +291,25 @@ public void flushAll() {
 		}
 	}
 
+	@Override
+	public void flushAllLocal() {
+		for (ResultSubpartition localSubpartition : localSubpartitions) {
+			localSubpartition.flush();
+		}
+	}
+
+	@Override
+	public void setFlushTimeout(long flushTimeout) {
+		checkState(!this.flushTimeout.isPresent(), "Flush timeout can not be set twice");
+		for (ResultSubpartition subpartition: remoteSubpartitionsMissingPeriodicFlushes) {
+			checkState(subpartition.isLocal().isPresent());
+			checkState(!subpartition.isLocal().get());
+			subpartition.registerPeriodicFlush(flushTimeout);
+		}
+		remoteSubpartitionsMissingPeriodicFlushes.clear();
```

Comment:
There may exit race condition between `setFlushTimeout()` and `createSubpartitionView()`?
If the task thread invokes `setFlushTimeout` and clears the `remoteSubpartitionsMissingPeriodicFlushes`, at the same time, the netty thread invokes creating subpartition view and adding this subpartition to the `remoteSubpartitionsMissingPeriodicFlushes` before setting `flushTimeout`.

Reply:
Yes, we may need add the synchronizing in above three methods for protecting  `localSubpartitions`, `remoteSubpartitionsMissingPeriodicFlushes` and `flushTimeout` which are operated by task thread, netty thread and flusher thread.

Reply:
In that case I would have to re-evaluate this. Adding extra synchronisation would further complicate the code and could cause regressions.